catalog/async/catalog-async-data/src/test/groovy/org/codice/ddf/catalog/async/data/LazyProcessResourceImplTest.groovy

```diff
@@ -0,0 +1,223 @@
+import spock.lang.Shared
+import spock.lang.Specification
+
+import java.util.function.Supplier
+
+import static org.codice.ddf.catalog.async.data.impl.ProcessResourceImpl.*
+
+class LazyProcessResourceImplTest extends Specification {
+
+    @Shared
+    String RESOURCE_NAME = "resourceName"
+
+    @Shared
+    int RESOURCE_SIZE = 1
+
+    @Shared
+    InputStream RESOURCE_INPUTSTREAM = Mock(InputStream)
+
+    @Shared
+    String RESOURCE_MIMETYPE = "mimeType"
+
+    @Shared
+    String METACARD_ID = "metacardId"
+
+    @Shared
+    URI RESOURCE_URI = new URI(CONTENT_SCHEME, METACARD_ID, null)
+
+    @Shared
+    String RESOURCE_QUALIFIER = ""
+
+
+    private Resource resource
+    private Supplier<Resource> supplier
+
+    def setup() {
+        resource = Mock(Resource) {
+            getSize() >> RESOURCE_SIZE
+            getInputStream() >> RESOURCE_INPUTSTREAM
+            getName() >> RESOURCE_NAME
+            getMimeTypeValue() >> RESOURCE_MIMETYPE
+        }
+
+        supplier = Mock(Supplier) {
+            get() >> resource
+        }
+    }
+
+
+    def 'new resource IllegalArgument on bad inputs'() {
+        when:
+        new LazyProcessResourceImpl(metacardId, resourceSupplier)
+
+        then:
+        thrown(IllegalArgumentException)
+
+        where:
+        metacardId   | resourceSupplier
+        null         | Mock(Supplier)
+        ""           | Mock(Supplier)
+        " "          | Mock(Supplier)
+        METACARD_ID | null
+
+    }
+
+    def 'new resource verifying lazy loading not loaded'() {
+        when:
+        new LazyProcessResourceImpl(METACARD_ID, supplier)
+
+        then:
+        0 * supplier.get()
+    }
+
+    def 'verify a new resource is lazy loaded at first get of resource specific field'() {
+        given:
+        Supplier<Resource> supplier = Mock(Supplier) {
+            1 * get() >> resource
+        }
+
+        long expectedSize = 3914
+        String uriString = "something:3839ab393df930303#frag"
+        URI expectedUri = new URI(uriString);
+        String expectedQualifier = "frag"
+
+        def lazyProcessResource = new LazyProcessResourceImpl(METACARD_ID, supplier)
+        lazyProcessResource.setSize(expectedSize)
+        lazyProcessResource.setUri(expectedUri)
+
+
+
+        when:
+        lazyProcessResource.getName()
+        lazyProcessResource.getMimeType()
+        lazyProcessResource.getName()
+        lazyProcessResource.close()
+
+        then:
+        lazyProcessResource.size == expectedSize
+        lazyProcessResource.uri == expectedUri
+        lazyProcessResource.qualifier == expectedQualifier
+        lazyProcessResource.name == RESOURCE_NAME
+        lazyProcessResource.inputStream == RESOURCE_INPUTSTREAM
+        lazyProcessResource.mimeType == RESOURCE_MIMETYPE
+        !lazyProcessResource.modified
+    }
+
+    def 'verify a new resource is not lazy loaded when accessing size, uri, or qualifier'() {
```

Comment:
Why should `getSize()`, `getUri()`, and `getQualifier()` not load the `Resource`? Details should be included in the javadoc for those methods or even in comments above this test method.

Reply:
Added comment to the test and to the javadoc.

Size, URI, and qualifier are stored in the metacard and know at the time of creating the LazyProcessResource.
They are set from the metacard values.

MimeType, name, and inputStream can only be populated once the resource has been loaded.